### PR TITLE
ENTSWM-178 Add tests for archaius fraction

### PR DESCRIPTION
**Pull request by juangon**

### CODE REVIEW

File: netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        archive.addAsWebInfResource(new File("src/main/resources/project-defaults.yml"));
+        return archive;
+    }
+
+    @Test
+    public void test() throws Exception {
+        Executor executor = Executor.newInstance().use(new BasicCookieStore());
+
+        String response = executor.execute(Request.Get("http://localhost:8080/")).returnContent().asString();
+        assertThat(response).isEqualTo("50");
```
Comment (Ladicek) [inline netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java]: I'd expect 77 here :-)

**Comment by Ladicek** [comment]
Why did you close this? This looks exactly like what I'd expect.

File: netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java (hunk 1 above)
Comment (juangon) [reply netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java]: Yeah, that's the reason I closed the pull :D

Comment (Ladicek) [reply netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java]: Ah so it's returning 50 now? Weird.

Comment (juangon) [reply netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java]: Yeah,  that is the default value from fraction. Probably it isn't getting the yaml file for some reason. Will investigate that on thursday...

Comment (Ladicek) [reply netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java]: OK cool.

**Comment by Ladicek** [comment]
I reopened, because this is valid, just needs more work, so won't merge for now.

**Comment by juangon** [comment]
@Ladicek pushed changes that makes it work now

File: netflix/archaius/pom.xml [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+            <groupId>org.assertj</groupId>
+            <artifactId>assertj-core</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.httpcomponents</groupId>
+            <artifactId>httpclient</artifactId>
+            <scope>test</scope>
+        </dependency>
+                <dependency>
```
Comment (Ladicek) [inline netflix/archaius/pom.xml]: Formatting :-)

File: netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    public static Archive<?> deployment() {
+        JAXRSArchive archive = ShrinkWrap.create(JAXRSArchive.class);
+        archive.addClass(ArchaiusResource.class).addClass(RestApplication.class);
+        archive.addAsResource(new File("src/main/resources/project-defaults.yml"));
+        return archive;
+    }
+
+    @Test
+    public void test() throws Exception {
+        Executor executor = Executor.newInstance().use(new BasicCookieStore());
```
Comment (Ladicek) [inline netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java]: You don't need to create the executor in this case -- this test doesn't rely on cookies at all. Better to just `Request.Get("...").execute().returnContent()...`

File: netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+@RunWith(Arquillian.class)
+public class ArchaiusTest {
+
+    @Deployment(testable = false)
+    public static Archive<?> deployment() {
+        JAXRSArchive archive = ShrinkWrap.create(JAXRSArchive.class);
+        archive.addClass(ArchaiusResource.class).addClass(RestApplication.class);
+        archive.addAsResource(new File("src/main/resources/project-defaults.yml"));
+        return archive;
```
Comment (Ladicek) [inline netflix/archaius/src/test/java/org/wildfly/swarm/ts/netflix/archaius/ArchaiusTest.java]: Making it a `<packaging>war</packaging>` in `pom.xml` and using `@DefaultDeployment` in the test didn't work?

**Comment by juangon** [comment]
@Ladicek pushed changes with latest suggestions.

**Comment by Ladicek** [comment]
Thanks!

**Comment by Ladicek** [comment]
Actually you forgot to include the module in root POM, but I did that now: https://github.com/Ladicek/swarm-test-suite/commit/010f2ba43207aa3b6b616a7135461ed755dba44f

**Pull request merged**